### PR TITLE
Use pdata for prometheus monitor

### PR DESCRIPTION
**Pull request by atoulme**

### CODE REVIEW

File: internal/signalfx-agent/pkg/monitors/prometheusexporter/conversion.go [hunk 1]
```diff
@@ -96,27 +138,51 @@ func makeHistogramDatapoints(name string, ms []*dto.Metric) []*datapoint.Datapoi
...
-			dps = append(dps, sfxclient.Cumulative(name+"_count", dims, int64(h.GetSampleCount()))) //nolint:gosec
+			metric := pmetric.NewMetric()
+			metric.SetName(name + "_count")
+			sum := metric.SetEmptySum()
+			sum.SetIsMonotonic(true)
+			sum.SetAggregationTemporality(pmetric.AggregationTemporalityCumulative)
+			dp := sum.DataPoints().AppendEmpty()
+			dp.SetIntValue(int64(h.GetSampleCount())) //nolint:gosec // disable G115
+			_ = dp.Attributes().FromRaw(dims)
+			dps = append(dps, metric)
```
Comment (dmitryax) [inline internal/signalfx-agent/pkg/monitors/prometheusexporter/conversion.go:149]: So, we are creating one metric with one data point for each Prometheus time series. Is that correct? I don't see any combining logic down the road...

Comment (dmitryax) [reply internal/signalfx-agent/pkg/monitors/prometheusexporter/conversion.go:149]: Looks like the existing `sfxDatapointsToPDataMetrics` implementation does pretty much the same though...

**Comment by dmitryax** [comment]
Do we have any tests to confirm that the output hasn't changed?

**Review by dmitryax — APPROVED** [review]
LGTM assuming we have tests covering the output

File: internal/signalfx-agent/pkg/monitors/prometheusexporter/conversion.go (hunk 1 above)
Comment (atoulme) [reply internal/signalfx-agent/pkg/monitors/prometheusexporter/conversion.go:149]: Yes, this is a simple follow through to make sure we get the same behavior for now.

**Comment by atoulme** [comment]
Yes, we have multiple tests that check with golden files that the prometheus metrics match. Both for simple gauges and counters and histograms.

**Pull request merged**